docs/source/reference/data.rst

```diff
@@ -389,17 +397,65 @@ require a dedicated way of being serialized.
 Our replay buffer enjoys two separate APIs for saving their state on disk:
 :meth:`~torchrl.data.ReplayBuffer.dumps` and :meth:`~torchrl.data.ReplayBuffer.loads` will save the
 data of each component except transforms (storage, writer, sampler) using memory-mapped
-tensors and json files for the metadata. This will work across all classes except
+tensors and json files for the metadata.
+
+This will work across all classes except
 :class:`~torchrl.data.replay_buffers.storages.ListStorage`, which content
 cannot be anticipated (and as such does not comply with memory-mapped data
 structures such as those that can be found in the tensordict library).
+
 This API guarantees that a buffer that is saved and then loaded back will be in
 the exact same state, whether we look at the status of its sampler (eg, priority trees)
 its writer (eg, max writer heaps) or its storage.
-Under the hood, :meth:`~torchrl.data.ReplayBuffer.dumps` will just call the public
+
+Under the hood, a naive call to :meth:`~torchrl.data.ReplayBuffer.dumps` will just call the public
 `dumps` method in a specific folder for each of its components (except transforms
 which we don't assume to be serializable using memory-mapped tensors in general).
 
+Saving data in TED-format may however consume much more memory than required. If continuous
+trajectories are stored in a buffer, we can avoid saving duplicated observations by saving all the
+observations at the root plus only the last element of the `"next"` sub-tensordict's observations, which
+can reduce the storage consumption up to two times. To enable this, three checkpointer classes are available:
+:class:`~torchrl.data.FlatStorageCheckpointer` will discard duplicated observations to compress the TED format. At
+load time, this class will re-write the observations in the correct format. If the buffer is saved on disk,
+the operations executed by this checkpointer will not require any additional RAM.
+The :class:`~torchrl.data.NestedStorageCheckpointer` will save the trajectories using nested tensors to make the data
+representation more apparent (each item along the first dimension representing a distinct trajectory).
+Finally, the :class:`~torchrl.data.H5StorageCheckpointer` will save the buffer in an H5DB format, enabling users to
+compress the data and save some more space.
+
+.. warning:: The checkpointers make some restrictive assumption about the replay buffers. First, it is assumed that
+  the ``done`` state accurately represents the end of a trajectory (except for the last trajectory which was written
+  for which the writer cursor indicates where to place the truncated signal). Furthermore, only done states that have
+  as many elements as the root tensordict are allowed: if the done state has extra elements that are not represented in
+  the batch-size of the storage, these checkpointers will fail.
```

Comment:
> Furthermore, only done states that have
  as many elements as the root tensordict are allowed: if the done state has extra elements that are not represented in
  the batch-size of the storage, these checkpointers will fail.
  
 Hmm did not quite catch this part. Is this referring to some special feature of the data format or tensordicts? If so, a link to this feature would be helpful.

Reply:
It's mostly for multi-agent stuff where they sometimes have done states with more elements than the batch size. Most people should not worry about that

Reply:
Ok, maybe worth just mentioning that this happens with multi-agent setup (or other examples where it can fail, or alternatively the setup where it works)? The wording was bit mouthful to understand :D